### PR TITLE
Stripe (again)

### PR DESCRIPTION
**Pull request by Gregoor**
The Kuma partner PR is here: https://github.com/mdn/kuma/pull/7887

I'll document things over in the Kuma PR since the implementation here is only for having a frontend to test it with and won't be enabled for end-users in any form. So I hope things here stay fairly uncontroversial. Will still need a review though!

### CODE REVIEW

File: build/constants.js [hunk 1]
```diff
@@ -71,6 +71,9 @@ const HOMEPAGE_FEED_DISPLAY_MAX = JSON.parse(
   process.env.BUILD_HOMEPAGE_FEED_DISPLAY_MAX || "5"
 );
 
+// If you want to enable this on local development use http://localhost.org:8000/api/v1/subscriptions/config/
+const SUBSCRIPTION_CONFIG_URL = process.env.SUBSCRIPTION_CONFIG_URL || null;
```
Comment (peterbe) [inline build/constants.js:75]: Just so I don't forget, this needs to be mentioned in `docs/envvars.md`. 
And if I may be nit picky, I think you should prefix it to:
```suggestion
const SUBSCRIPTION_CONFIG_URL = process.env.BUILD_SUBSCRIPTION_CONFIG_URL || null;
```
so it matches the other expected env vars in this file.

**Review by peterbe — COMMENTED** [review]
This PR is still a draft but it works fine and should be good to go. 
The only nits are:

1. The name of the env var should have `BUILD_` prefix
2. The `BUILD_SUBSCRIPTION_CONFIG_URL` should be mentioned in `docs/envvars.md`

**Comment by Gregoor** [comment]
didelidone and also added a more explicit error if it fails despite a configured URL (like in my case where I did not have kuma running)

File: build/spas.js [hunk 2]
```diff
@@ -37,6 +40,23 @@ async function buildSPAs(options) {
...
+        await got(BUILD_SUBSCRIPTION_CONFIG_URL, { responseType: "json" })
+      ).body;
+    } catch (error) {
+      console.error(
+        "Error while fetching subscription config for",
+        BUILD_SUBSCRIPTION_CONFIG_URL,
+        ":",
+        error
+      );
+      throw error;
```
Comment (peterbe) [inline build/spas.js:56]: 👍

File: client/src/settings/subscription.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      {error && <div className="notecard negative">{error.toString()}</div>}
+      {current ? (
+        <>
+          <button disabled={isLoading} onClick={() => manage()}>
+            Manage
+          </button>
+          <pre>{JSON.stringify(current, null, 2)}</pre>
+        </>
+      ) : (
+        <button disabled={isLoading} onClick={() => checkout()}>
```
Comment (peterbe) [inline client/src/settings/subscription.tsx:96]:
```suggestion
        <button disabled={isLoading} onClick={() => checkout(config.prices[0])}>
```
...or...
```suggestion
        <button disabled={isLoading} onClick={() => checkout(config.prices[0].id)}>
```
and then, pass it on into `redirectToStripeCheckout` etc. You get the idea :)

I know it's early days for this code, and how the buttons work is going to greatly change, but I think the `redirectToStripeCheckout` is solid and can stay but I just don't like how it hardcodes it to `prices[0]` there.

File: docs/envvars.md [hunk 4]
```diff
@@ -191,6 +191,14 @@ Which RSS feed URL to parse for displaying feed entries on the home page.
 
 How many RSS feed entries to display on the home page.
 
+### `BUILD_SUBSCRIPTION_CONFIG_URL`
+
+**Default: `'''``**
```
Comment (peterbe) [inline docs/envvars.md:196]:
```suggestion
**Default: `''`**
```

**Review by peterbe — APPROVED** [review]
With nits...

File: client/src/settings/subscription.tsx (hunk 3 above)
Comment (Gregoor) [reply client/src/settings/subscription.tsx:96]: I think I will keep it as is for now, as I would prefer the current version if there is only one price. And I think there is still a chance we're going that way. So I will respectfully ignore it!

**Pull request merged**